src/main/java/ch/puzzle/eft/controller/ExamDownloadController.java

```diff
@@ -1,13 +1,35 @@
 package ch.puzzle.eft.controller;
 
+import ch.puzzle.eft.service.ExamFileService;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.ResponseBody;
+import java.io.File;
 
 @RestController
 public class ExamDownloadController {
+    ExamFileService examFileService;
 
-    @GetMapping("/download/")
-    public void downloadExam() {
-        // Download exam
+    public ExamDownloadController(ExamFileService examFileService) {
+        this.examFileService = examFileService;
+    }
+
+
+    @GetMapping(value = "/download/{subject}/{fileName}", produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
```

Comment:
MediaType.MediaType.APPLICATION_OCTET_STREAM_VALUE seems to be an unknown MIME-Type that is used for files that don't fall under any other MIME-Type.
See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#applicationoctet-stream
There seems to be a MIME type for PDF files, so try if you can use that.
https://stackoverflow.com/questions/312230/proper-mime-media-type-for-pdf-files

Reply:
Used MediaType.APPLICATION_PDF_VALUE